sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -1810,8 +1876,40 @@ pi_result piTearDown(void *) {
       PiESimdDeviceAccess->data);
   delete PiESimdDeviceAccess;
 
-  const std::lock_guard<sycl::detail::SpinLock> Lock{*PiESimdSurfaceMapLock};
   for (auto it = PiESimdSurfaceMap->begin(); it != PiESimdSurfaceMap->end();) {
```

Comment:
why the lock is removed? Is piTearDown guaranteed to be called from single thread? If so, please add comment

Reply:
Initially I asked for clarification why some mutexes are needed, in the next update mutexes have been removed.
Regarding `piTearDown`, probably it's not stated explicitly, but AFAIK the current assumption is that init and teardown APIs are guaranteed to be called only once(and hence in thread safe manner).

Reply:
@dongkyunahn-intel,  please add a comment then, as requested above, that that is it clear to the reader why it is safe.

Reply:
`piPluginInit` and `piTearDown` are guaranteed to be called only once for plug-in.